src/vehicle.cpp

```diff
@@ -3504,35 +3383,124 @@ int vehicle::charge_battery (int amount, bool include_other_vehicles)
     return amount;
 }
 
-int vehicle::discharge_battery (int amount, bool recurse)
+int vehicle::discharge( int amount, bool recurse, bool reactor )
 {
+    // try initially to obtain power from local batteries
     for( auto &p : parts ) {
         if( amount <= 0 ) {
             break;
         }
-        if( !p.is_broken() && p.is_battery() ) {
+        if( p.is_battery() && !p.is_broken() ) {
             amount -= p.ammo_consume( amount, global_part_pos3( p ) );
         }
     }
 
-    auto discharge_visitor = [] (vehicle* veh, int amount, int lost) {
-        g->u.add_msg_if_player(m_debug, "CH: %d", amount + lost);
-        return veh->discharge_battery(amount + lost, false);
-    };
-    if(amount > 0 && recurse) { // need more power!
-        amount = traverse_vehicle_graph(this, amount, discharge_visitor);
+    // if insufficient local power try any networked vehicles
+    if( amount > 0 && recurse ) {
+        amount = traverse_vehicle_graph( this, amount, []( vehicle *veh, int amount, int lost ) {
+            return veh->discharge( amount + lost, false, false );
+        } );
+    }
+
+    // if insufficient local or network electrical power try to engage any reactors
+    if( amount > 0 && reactor ) {
+        for( auto &pt : parts ) {
+
+            // consider only enabled reactors that have remaining fuel
+            if( pt.is_reactor() && pt.enabled && !pt.is_broken() && pt.ammo_remaining() ) {
+
+                const itype *fuel = item::find_type( pt.ammo_current() );
+                if( fuel->ammo ) {
```

Comment:
This should be enforced somewhere else - if all reactors need ammo-type ammo, then for a reactor, `ammo_remaining() > 0` should imply `item::find_type( pt.ammo_current() )->ammo != nullptr`.